### PR TITLE
[hotfix] fix dist spec mgr

### PR DESCRIPTION
**Pull request by ver217**

### CODE REVIEW

File: colossalai/tensor/dist_spec_mgr.py [hunk 1]
```diff
@@ -34,7 +34,7 @@ def _shard_as(tensor: torch.Tensor, old_dist_spec: _DistSpec, dist_spec: _DistSp
             chunk_size = divide(tensor.size(dim), dist_spec.num_partitions[i])
             chunk = chunk.narrow(dim, idx // num_parts * chunk_size, chunk_size)
             idx %= num_parts
-        return chunk.detach().contiguous()
+        return chunk.clone().detach().contiguous()
```
Comment (feifeibear) [inline colossalai/tensor/dist_spec_mgr.py:37]: _shard_as is only called on model data before iteration, right?

Comment (ver217) [reply colossalai/tensor/dist_spec_mgr.py:37]: It can also be called on activation, but it's not in-place operation for activation.

Comment (feifeibear) [reply colossalai/tensor/dist_spec_mgr.py:37]: then clone will introduce non-trivial overhead. especially clone on CPU.

Comment (ver217) [reply colossalai/tensor/dist_spec_mgr.py:37]: But without `clone()`, there may be side-effect. E.g. user modifies the original tensor after `_shard_as()`, then the content shard is modified as well.

**Review by feifeibear — APPROVED** [review]

**Pull request merged**